.github/workflows/linting.yml

```diff
@@ -0,0 +1,19 @@
+name: linting
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+          cache: pip
+      - name: Lint the code
+        uses: pre-commit/action@v2.0.0
```

Comment:
fyi - these days its recommended to use pre-commit.ci instead of the action

its faster to run and prevents a few pain points of the action

Reply:
Yes, agree. I've removed `linting.yml`

Reply:
Should I add the repo to `pre-commit.ci`?

Reply:
Yes

Reply:
It just added!

Reply:
It uses the configuration in `.pre-commit-config.yaml` file

Reply:
Yes but there's nothing in the configuration file to tell pre-commit.ci that it should run on this repo 😕

Reply:
I've added the repo to pre-commit.ci and it handles the rest

Reply:
I guess what I'm asking is, what exactly do you mean by "added the repo to pre-commit.ci"?

Reply:
@hramezani Just bumping this discussion in case you missed it, I think this is the last thing I'd like to understand before I'll be on board with merging the PR

Reply:
I am not sure that I get your question right.
What I meant by adding the repo to pre-commit.ci is to add the repository in pre-commit.ci website. you need to login first and add the repo. by adding the repo you give some access to the website. Then github runs the pre-commit hook on each and every pull request

Reply:
Ah, I see what you mean. I didn't realize you could log into the website or register a repository there. I just hope that if we ever want to turn off the pre-commit integration, somebody remembers that we need to go do it on their website. It seems pretty unintuitive to me :shrug:

Anyway, this is good enough for me to proceed.